### PR TITLE
Optimize and simplify kernel

### PR DESCRIPTION
**Pull request by JoshVorick**

### CODE REVIEW

File: sia-gpu-miner.c [hunk 1]
```diff
@@ -71,47 +77,43 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
...
 		fflush(stdout);
+		return -1;
 	}
 	target_corrupt_flag = 0;
+	size_t global_item_size = 1 << intensity;
+
+	// Copy target to header
+	for (i = 0; i < 8; i++) {
+		blockHeader[i + 32] = target[7-i];
+	}
```
Comment (lukechampine) [inline sia-gpu-miner.c:90]: `memcpy`?

Comment (JoshVorick) [inline sia-gpu-miner.c:90]: Can't memcopy because I'm copying the target backwards because GPU uses little endian

Comment (lukechampine) [inline sia-gpu-miner.c:90]: ah, gotcha. Shouldn't you be copying 32 bytes though, not 8?

Comment (DavidVorick) [inline sia-gpu-miner.c:90]: it's an optimization. Only the first 6 bytes of the target matter, and the other 2 are a buffer. We won't need more than that unless the hashrate increases by about 100,000x the current hashrate, at which point we'll just switch from looking at the first 8 bytes to looking at bytes 4-12.

**Pull request merged**